functions.php

```diff
@@ -264,18 +266,44 @@ function( $field ) {
 
 		$first_name = rgar( $entry, '' . $first_name_field->id . '.3' );
 
-		$post = Timber::query_post( false, Post::class );
-
 		$confirmation_fields = [
 			'first_name'       => $first_name ?? '',
 			'post'             => $post,
 			'form_title'       => $form['title'] ?? '',
 			'form_description' => $form['description'] ?? '',
 		];
 
-		return Timber::compile( [ 'gravity_forms_confirmation.twig' ], $confirmation_fields );
+		return Timber::compile( [ 'gravity_forms_p4_confirmation.twig' ], $confirmation_fields );
 	} else {
-		return $confirmation;
+		// If the $confirmation object is an array, it means that it's a redirect page so we can directly use it.
+		if ( is_array( $confirmation ) ) {
+			return $confirmation;
+		}
+
+		$default_confirmation_array = array_filter(
```

Comment:
There is probably a better way to find the current confirmation message, at first I wanted to check for the `isDefault` field but since editors can add other confirmation messages and show them based on conditional logic, I thought this made more sense... But if the message string is the same in all confirmation messages, then this will not work properly! I'm not sure what's best here 😬